components/PostingLocation/SuggestionsRooms/SuggestionsRooms.jsx

```diff
@@ -47,6 +48,24 @@ export class SuggestionsRooms extends Component {
     }
   }
 
+  removeDuplicateFilter() {
+    let _filters = this.props.filters;
+
+    if(_filters.length > 0) {
+      _filters.sort();
+      for(let i = 0, n = _filters.length; i < n-1; i++) {
```

Comment:
You might loop the array using 'filter' rather than 'for loop' and use hashtables to check the duplication.

Something like that:
```
    let seen = {};
    let uniqueArray = _filters.filter((item) => {
        return seen.hasOwnProperty(item.threadId) ? false : (seen[item.threadId] = true);
    });
    this.setState({
        filters: _filters,
    });
```